walg_archive.c

```diff
@@ -64,6 +64,16 @@ _PG_archive_module_init(ArchiveModuleCallbacks *cb)
 	cb->archive_file_cb = walg_archive_file;
 }
 
+
+/*
+ * We use frontend/backend protocol to communicate through UNIX-socket
+ * So we define message format as array of bytes:
+ * 1 byte - type of message, char
+ * 1 byte - (N) len of message body including itself, int8
+ * (N - 1) byte - message body, char
```

Comment:
The length of the message should include the header length, so the message body length would be (N-3) bytes.